LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -8,20 +8,29 @@ namespace LoRaWan.NetworkServer
     using System;
     using System.Buffers.Binary;
     using System.Security.Cryptography;
-    using System.Text;
-    using BasicsStation;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using LoRaTools.LoRaMessage;
     using Microsoft.Extensions.Caching.Memory;
     using Microsoft.Extensions.Logging;
 
-    internal sealed class ConcentratorDeduplication<T> :
-        IConcentratorDeduplication<T>, IDisposable
-        where T : class
+    public sealed class ConcentratorDeduplication :
+        IConcentratorDeduplication, IDisposable
     {
+        public enum Result
+        {
+            NotDuplicate,
+            DuplicateDueToResubmission,
+            SoftDuplicate, // detected as a duplicate but due to the DeduplicationStrategy marked as a "soft" duplicate
```

Comment:
nit: not sure if we can come up with a better name. I'd like to have information about the fact that this is a concentrator related duplicate, but we let it go, because the dedup strategy is mark or none. Maybe ConcentratorDuplicatePass and then for the `Duplicate` ConcentratorDuplicateDrop? Any other thoughts?

Reply:
What about `SoftDuplicateDueToDeduplicationStrategy`? I would avoid including "allow" or "drop" on the name because imo this service should only be concerned about the detection part and not with what will happen with the message. The decision to drop or allow depends on the caller and the semantics there could change in the future.

Reply:
Changed to above suggestion in 94dbca2198ae7bf53718dba65f248cd5af317d54